homeassistant/components/tuya/binary_sensor.py

```diff
@@ -293,10 +302,58 @@ def __init__(
         self.entity_description = description
         self._attr_unique_id = f"{super().unique_id}{description.key}"
 
-    @property
-    def is_on(self) -> bool:
-        """Return true if sensor is on."""
-        dpcode = self.entity_description.dpcode or self.entity_description.key
-        if dpcode not in self.device.status:
-            return False
-        return self.device.status[dpcode] == self.entity_description.on_value
+    @callback
+    def async_off_delay_handler(self, _: datetime) -> None:
+        """Switch binary sensor off after a delay."""
+        self._off_delay_listener = None
+        self._attr_is_on = False
+        self.async_write_ha_state()
+
+    @callback
+    def _async_handle_update(self) -> None:
+        """Handle when payload is received for this device."""
+        self._attr_is_on = False
+        if (
+            dpcode := self.entity_description.dpcode or self.entity_description.key
+        ) in self.device.status:
+            self._attr_is_on = (
+                self.device.status[dpcode] == self.entity_description.on_value
+            )
+
+            # In case of an off delay, the device will not report the state again
+            # with a different or off value.
+            if self.entity_description.off_delay:
+                if self._off_delay_listener is not None:
+                    self._off_delay_listener()
+                    self._off_delay_listener = None
+
+                if self._attr_is_on:
+                    self._off_delay_listener = async_call_later(
```

Comment:
I don't like this. We're faking state which is something we're always declining otherwise.

Reply:
For motion/presence sensor that is not true, ZHA and MQTT are examples that have this. In our recent templates refactor the same is in.

I this was a button push, I would 100% agree. For motion/presence sensors; this is a common pattern.

Reply:
Regardless of if there are existing bad patterns, I don't think we should add more.

We made the template work so that users can easily make triggers into binary sensor, yes. If tuya doesn't report an on/off state here it should just be an event.

Reply:
> We made the template work so that users can easily make triggers into binary sensor, yes

That isn't easy of user friendly, we consider YAML to be advanced.

I'll leave it broken. Don't care.